### PR TITLE
[docs] User tutorial

### PR DESCRIPTION
**Pull request by francishemingway**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [x] Documentation

### SUMMARY
A new tutorial that focusing more on how an end user might use Superset, rather than how an administrator might set it up and get it working for the first time. I have added this as a new rst file, and changed the hierarchy within the documents so that there's now a tutorials page, with links to the existing tutorial and the new one.

The tutorial starts with basic instructions on how to load CSV data into superset and create a visualization. It then progresses through some different visualizations (pivot, bar chart, line chart), dashboard elements (filters, markdown and publishing). Then, it covers some more advanced features: advanced analytics and annotations.

Here's a screenshot of the new hierarchy of tutorials:
image

Full text of the tutorial is [here](https://docs.google.com/document/d/1T6Zy9slc1vHHIGp8jepnMjplf2nMHxLTY-SyORh9hUE/edit?usp=sharing) (you should be able to comment).

### ADDITIONAL INFORMATION
This tutorial is based on a real dataset from [here](https://data.gov.uk/dataset/b2bc9c3f-c892-4bcf-b8c1-0d0efdace3fa/dft-flight-data) which is a set of flights taken by an organization in 2011. This data is licensed under the Open Government Licence v3.0, an ASF category A license. Then it was combined with data from [here](https://ourairports.com/data/), licensed under the public domain. Then I tidied it up by removing a single line (a flight with the same origin and destination). It's currently [on my github](https://github.com/francishemingway/unofficial-superset-tutorial/blob/master/tutorial_flights.csv).

### Questions I have:
It would be great to get some guidance on the following:
- Is there somewhere within the project's repositories to host the data file.
- Should I open a new issue? This addresses more fully  #698.

### CODE REVIEW

**Comment by etr2460** [comment]
This looks awesome, would you mind attaching some screenshots of the new documentation to make it a bit easier to review?

also, @sylvia-tomiyama you might be interested in this

**Comment by francishemingway** [comment]
@etr2460 , @sylvia-tomiyama 
I have added a screenshot to show the new organization of tutorials. Below that I've linked to the text in a google doc (it's about 2,500 words, so it's not going to fit in a screenshot). You _should_ be able comment - let me know if not.

**Review by willbarrett — CHANGES_REQUESTED** [review]
I saw a few minor language tweaks that could be good, my only blocker is moving the data to the already-recommended repo. Thank you very much for doing this!

File: docs/usertutorial.rst [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+- Whether the ticket was a single or return.
+- The date of travel.
+- Information about the origin and destination.
+- The distance between the origin and destination, in km.
+
+Obtaining and loading the data
+------------------------------
+
+Download the data for this tutorial to your computer from `Github
+<https://github.com/francishemingway/unofficial-superset-tutorial/blob/master/tutorial_flights.csv>`_.
```
[the remark this answers is not in the document]
Comment (francishemingway) [reply docs/usertutorial.rst]: I have updated the link, but will keep this open until the other pull request has been merged.

**Comment by francishemingway** [comment]
@willbarrett - thanks very much for your feedback!

@etr2460 - I have created a pull request in the other repo [here](https://github.com/apache-superset/examples-data/pull/2) and I've updated the URL in this pull request to reflect where the new file will be.

File: docs/usertutorial.rst [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+- Information about the origin and destination.
+- The distance between the origin and destination, in kilometers (km).
+
+Obtaining and loading the data
+------------------------------
+
+Download the data for this tutorial to your computer from `Github
+<https://raw.githubusercontent.com/apache-superset/examples-data/master/tutorial_flights.csv>`_.
+
+In the top menu, select **Sources > Upload a CSV**.
```
Comment (dpgaspar) [inline docs/usertutorial.rst]: By default no databases are allowed for CSV upload. Add a step to show how to change the examples DB connection or create the `upload_csv` db connection

File: docs/usertutorial.rst [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+
+.. image:: images/usertutorial/upload_a_csv.png
+
+Then, enter the **Table name** as tutorial_flights and select the **CSV file**
+from your computer.
+
+.. image:: images/usertutorial/csv_to_database_configuration.png
+
+Next enter the text Travel Date into the **Parse Dates** field.
```
Comment (dpgaspar) [inline docs/usertutorial.rst]: nit: change  Travel Date to `Travel Date`

File: docs/usertutorial.rst [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+we want to look at all the data in the dataset. No problem - within the
+**Time** section, remove the filter on **Time range** by selecting on
+**Last week** **> No filter**, with a final **OK** to confirm your
+selection.
+
+.. image:: images/usertutorial/no_filter_on_time_filter.png
+
+Now, we want to specify the rows in our table by using the **Group by**
+option. Since in this example, we want to understand different Travel
+Classes, we select Travel Class in this menu.
```
Comment (dpgaspar) [inline docs/usertutorial.rst]: nit: replace Travel Class by `Travel Class`

**Review by dpgaspar — CHANGES_REQUESTED** [review]

**Review by etr2460 — COMMENTED** [review]
The other PR is merged now, so once you address the other comments this should be good to go!

File: docs/usertutorial.rst (hunk 4 above)
Comment (francishemingway) [reply docs/usertutorial.rst]: I've done this using :guilabel: see [here for more info](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html)

image

File: docs/usertutorial.rst (hunk 3 above)
Comment (francishemingway) [reply docs/usertutorial.rst]: same applies as to the other comment on this topic

File: docs/usertutorial.rst (hunk 2 above)
Comment (francishemingway) [reply docs/usertutorial.rst]: I've added a section explaining how to change the examples database connection

**Comment by francishemingway** [comment]
@etr2460 I've updated based on all of the feedback now. Unless there's any more feedback, should be good to go.

**Review by etr2460 — APPROVED** [review]
lgtm, thanks for adding all this documentation!

**Review by willbarrett — APPROVED** [review]

**Pull request merged**